### ISSUE TITLE
Cannot change settings in dials.recprocal_lattice_viewer when the detector swing angle is larger than 90

### ISSUE THREAD

**Issue by biochem-fan** [dials/dials#2481]
When the detector swing angle (THETA in Rigaku/Oxford's terminology but more adequately called $2\theta$) is larger than 90 degrees, `dials.reciprocal_lattice_viewer` cannot find the beam centre and raises exceptions.

```
Traceback (most recent call last):
  File "/home/prog/dials-dev/modules/dials/src/dials/util/reciprocal_lattice/viewer.py", line 478, in OnChangeSettings
    self.parent.update_settings()
  File "/home/prog/dials-dev/modules/dials/src/dials/util/reciprocal_lattice/viewer.py", line 179, in update_settings
    self.set_beam_centre(self.settings.beam_centre_panel, self.settings.beam_centre)
  File "/home/prog/dials-dev/modules/dials/src/dials/util/reciprocal_lattice/__init__.py", line 293, in set_beam_centre
    raise ValueError("No detector intersection")
ValueError: No detector intersection

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/prog/dials-dev/modules/dials/src/dials/util/reciprocal_lattice/viewer.py", line 483, in OnChangeSettings
    self.beam_fast_ctrl.SetValue(old_beam_centre[0])
TypeError: 'NoneType' object is not subscriptable
```

Because of this, we cannot toggle what to plot (e.g. `all`, `indexed`, `unindexed` etc). Other toggles (e.g. `Show reciprocal cell`) work but are reflected only after forcing redraw by dragging the image by mouse.

(Another issue: when there are multiple experiments, which experiment do `rotation axis`, `beam vector` etc refer to?)

This is conceptually related to #2478.

## How to reproduce

Download `frames/exp_864_16_*.rodhypix` from [XRDa-0094](https://xrda.pdbj.org/entry/94), zero pad file names, import, spot find and index. Start `dials.reciprocal_lattice_viewer indexed.{expt,refl}` and tweak toggles.

## Potential solution

How about just `pass`ing, instead of raising an exception?
https://github.com/dials/dials/blob/9e7f4dc20e4d12fce6ac5af2f381e87a0942e8fe/src/dials/util/reciprocal_lattice/__init__.py#L289-L296

**Comment by dagewa** [comment]
I'm seeing different behaviour, but it is also bad (dials and dxtbx both on main).
```
dials.import renumbered/exp_864_16_0*
dials.find_spots imported.expt
dials.index imported.expt strong.refl
dials.rlv indexed.expt indexed.refl
```
Then click `indexed` button.

The lattice distorts and moves closer to the origin (green cross). Clicking back on `all` does not recover the correct geometry. No errors are printed. :thinking:

**Comment by dagewa** [comment]
Note, `dev.dials.napari_rlv` appears to work correctly in this case.

**Comment by biochem-fan** [comment]
@dagewa
Could you please try https://github.com/dials/dxtbx/commits/FormatROD-multiaxis branch of dxtbx? The old version (already merged) had the detector upside down, meaning that the detector normal was pointing the other way. My understanding is that the new code is correct, validated by positive anomalous peaks on lysozyme sulfurs.

**Comment by biochem-fan** [comment]
Old dxtbx:

```
Detector:
Panel:
  fast_axis: {-0.300409,-0.00215211,0.953808}
  slow_axis: {0.00213751,-0.999996,-0.00158311}
  origin: {44.0512,37.1859,-26.723}
  distance: 34.0465

Max resolution (at corners): 0.825281
Max resolution (inscribed):  1.781267

Beam:
    wavelength: 1.54056
    sample to source direction : {-0,-0.0020945,0.999998}

Goniometer:
    Rotation axis:   {0,-1,0}
    Fixed rotation:  {1,0,0,0,1,0,0,0,1}
    Setting rotation:{1,0,0,0,1,0,0,0,1}
```

New dxtbx:
```
Detector:
Panel:
  fast_axis: {-0.29113,-0.00280456,0.956679} 
  slow_axis: {0.00446379,0.999981,0.00428989}
  origin: {43.8402,-37.1446,-26.9241}
  distance: -34.3078

Max resolution (at corners): 0.825986
Max resolution (inscribed):  1.783561

Beam:
    wavelength: 1.54056
    sample to source direction : {-0,9.99172e-05,1}

Goniometer:
    Rotation axis:   {0,-1,0}
    Fixed rotation:  {-0.243636,-0.670805,-0.700473,0.11582,0.696945,-0.70771,0.962926,-0.253552,-0.0921083}
    Setting rotation:{1,0,0,0,1,0,0,0,1}
    Axis #0 (PHI):  {0,-1,0}
    Axis #1 (KAPPA=CHI):  {0,-0.641686,0.766967}
    Axis #2 (OMEGA):  {0,-1,0}
    Angles: 60,61,136
    scan axis: #2 (OMEGA)
```

Anyway, whatever the geometry, something is wrong with `dials.reciprocal_lattice_viewer`...

**Comment by dagewa** [comment]
I swapped to your branch, but I still see the distortion behaviour. Agree, something definitely wrong with `dials.rlv`
[Screencast from 01-08-23 09:45:29.webm](https://github.com/dials/dials/assets/11502083/66ca6cfe-ecf1-4599-bcea-cdd362ef00ad)

**Comment by biochem-fan** [comment]
Did you import images again after switching to the latest dxtbx?

I reproduced your distortion behavior if I import images using the old (merged) version. In this case, I see no exceptions but distortions as you reported.

**Comment by dagewa** [comment]
Yes, re-imported:
```
$ dials.show imported.expt 
DIALS (2018) Acta Cryst. D74, 85-97. https://doi.org/10.1107/S2059798317017235
The following parameters have been modified:

input {
  experiments = imported.expt
}

Experiment 0:
Experiment identifier: 8079e8f9-88ad-a5cc-823f-d16d47fabb34
Image template: /home/fcx32934/data/2478-fix/renumbered/exp_864_16_###.rodhypix
Detector:
Panel:
  name: Panel
  type: PAD
  identifier: 
  pixel_size:{0.1,0.1}
  image_size: {775,800}
  trusted_range: {0,51302}
  thickness: 0
  material: 
  mu: 0
  gain: 1
  pedestal: 0
  fast_axis: {-0.304867,-0.00228638,0.952392}
  slow_axis: {0.00073249,0.999996,0.00263513}
  origin: {44.2073,-36.9766,-26.7606}
  distance: -34
  pixel to millimeter strategy: SimplePxMmStrategy


Max resolution (at corners): 0.825285
Max resolution (inscribed):  1.776247

Beam:
    wavelength: 1.54056
    sample to source direction : {0,0,1}
    divergence: 0
    sigma divergence: 0
    polarization normal: {0,1,0}
    polarization fraction: 0.5
    flux: 0
    transmission: 1

Scan:
    number of images:   56
    image range:   {1,56}
    oscillation:   {136,0.5}
    exposure time: 0.05

Goniometer:
    Rotation axis:   {0,-1,0}
    Fixed rotation:  {-0.243636,-0.670805,-0.700473,0.11582,0.696945,-0.70771,0.962926,-0.253552,-0.0921083}
    Setting rotation:{1,0,0,0,1,0,0,0,1}
    Axis #0 (PHI):  {0,-1,0}
    Axis #1 (KAPPA=CHI):  {0,-0.641686,0.766967}
    Axis #2 (OMEGA):  {0,-1,0}
    Angles: 60,61,136
    scan axis: #2 (OMEGA)

```
It's all very puzzling.

BTW if I try to show the reciprocal cell _before_ clicking the `indexed` button, then I get a different error:
```
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
Traceback (most recent call last):
  File "/home/fcx32934/sw/cctbx/modules/cctbx_project/wxtbx/utils.py", line 38, in <lambda>
    lambda evt: self.update_values(evt, setting),
  File "/home/fcx32934/sw/cctbx/modules/cctbx_project/wxtbx/utils.py", line 76, in update_values
    self.parent.update_settings()
  File "/home/fcx32934/sw/cctbx/modules/dials/src/dials/util/reciprocal_lattice/viewer.py", line 179, in update_settings
    self.set_beam_centre(self.settings.beam_centre_panel, self.settings.beam_centre)
  File "/home/fcx32934/sw/cctbx/modules/dials/src/dials/util/reciprocal_lattice/__init__.py", line 279, in set_beam_centre
    panel = int(panel)
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

**Comment by biochem-fan** [comment]
Your "different error" seems what I initially described.
